packages/react-menu/src/utils/useOnMenuEnter.ts

```diff
@@ -36,12 +36,21 @@ export const useOnMenuMouseEnter = (options: UseOnClickOrScrollOutsideOptions) =
   });
 
   React.useEffect(() => {
+    // eslint-disable-next-line eqeqeq
+    if (element == null) {
+      return;
+    }
+
     if (!disabled) {
-      element?.addEventListener(MENU_ENTER_EVENT, listener);
+      element.addEventListener(MENU_ENTER_EVENT, evt => {
+        listener(evt as MouseEvent);
```

Comment:
`evt` callback type override falls back to 2nd definition (widened `string` for event name, thus the `evt` is being typed base class of MouseEvent -> `Event`

- this fixes it
- in general it's not a good idea to provide callback reference to native API's as that might break and TS wont trigger error (https://jakearchibald.com/2021/function-callback-risks/)